clients/src/main/java/org/apache/kafka/common/serialization/Serializations.java

```diff
@@ -0,0 +1,144 @@
+            return deserializer;
+        }
+    }
+
+    static public class IntegerSerialization implements Serialization<Integer> {
+        private Serializer<Integer> serializer = new IntegerSerializer();
+        private Deserializer<Integer> deserializer = new IntegerDeserializer();
+
+        @Override
+        public Serializer<Integer> serializer() {
+            return serializer;
+        }
+
+        @Override
+        public Deserializer<Integer> deserializer() {
+            return deserializer;
+        }
+    }
+
+    static public class StringSerialization implements Serialization<String> {
+        private Serializer<String> serializer = new StringSerializer();
+        private Deserializer<String> deserializer = new StringDeserializer();
+
+        @Override
+        public Serializer<String> serializer() {
+            return serializer;
+        }
+
+        @Override
+        public Deserializer<String> deserializer() {
+            return deserializer;
+        }
+    }
+
+    static public class ByteArraySerialization implements Serialization<byte[]> {
+        private Serializer<byte[]> serializer = new ByteArraySerializer();
+        private Deserializer<byte[]> deserializer = new ByteArrayDeserializer();
+
+        @Override
+        public Serializer<byte[]> serializer() {
+            return serializer;
+        }
+
+        @Override
+        public Deserializer<byte[]> deserializer() {
+            return deserializer;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    static public <T> Serialization<T> serialization(Class<T> type) {
+        if (String.class.isAssignableFrom(type)) {
+            return (Serialization<T>) STRING();
+        }
+
+        if (Integer.class.isAssignableFrom(type)) {
+            return (Serialization<T>) INT();
+        }
+
+        if (Long.class.isAssignableFrom(type)) {
+            return (Serialization<T>) LONG();
+        }
+
+        if (byte[].class.isAssignableFrom(type)) {
+            return (Serialization<T>) BYTEARRAY();
+        }
+
+        // TODO: we can also serializes objects of type T using generic Java serialization by default
+        throw new IllegalArgumentException("Unknown class for built-in serializer");
+    }
+
+    static public <T> Serialization<T> serialization(final Serializer<T> serializer, final Deserializer<T> deserializer) {
+        return new Serialization<T>() {
+            @Override
+            public Serializer<T> serializer() {
+                return serializer;
+            }
+
+            @Override
+            public Deserializer<T> deserializer() {
+                return deserializer;
+            }
+        };
+    }
+
+    /*
+     * A serde for nullable long type.
+     */
+    static public Serialization<Long> LONG() {
+        return new LongSerialization();
+    }
+
+    /*
+     * A serde for nullable long type.
+     */
+    static public Serialization<Integer> INT() {
```

Comment:
I'd rename this to `INTEGER`.  This reduces confusion when using a Scala wrapper, for example (`Int` in Scala is different from `Integer`).